src/V3MergeCond.cpp

```diff
@@ -407,7 +389,8 @@ class MergeCondVisitor final : public VNVisitor {
             m_mgFirstp = nodep;
             m_mgCondp = condp;
             m_listLenght = 0;
-            m_markVars.mark(condp);
+            // Mark variable references in the condition
+            condp->foreach<AstVarRef>([](const AstVarRef* nodep) { nodep->varp()->user1(1); });
```

Comment:
This is so much better to read!!

Reply:
BTW all these are inspired by the Scala standard library collections, which also have `exists`, `forall`, `collect`, etc. Sadly does not fit in C++ just as nicely.